### PR TITLE
Webcams only for moderator

### PR DESCRIPTION
**Pull request by GhaziTriki**
Only the moderator is able to see webcam streams, whether he shares his stream or receives other streams.

### CODE REVIEW

File: bigbluebutton-client/src/org/bigbluebutton/core/model/MeetingBuilder.as [hunk 1]
```diff
@@ -54,6 +55,11 @@ package org.bigbluebutton.core.model
       allowStartStopRecording = value;
       return this;
     }
+	
+	public function withwebcamsOnlyForModerator(value: Boolean):MeetingBuilder {
```
Comment (capilkey) [inline bigbluebutton-client/src/org/bigbluebutton/core/model/MeetingBuilder.as]: Webcam should be capitalized here.

File: bigbluebutton-client/src/org/bigbluebutton/main/model/users/JoinService.as [hunk 2]
```diff
@@ -160,7 +161,9 @@ package org.bigbluebutton.main.model.users
                                              .withExternalId(response.externMeetingID).withRecorded(response.record.toUpperCase() == "TRUE")
                                              .withDefaultAvatarUrl(response.avatarURL).withDialNumber(response.dialNumber)
                                              .withWelcomeMessage(response.welcome).withModOnlyMessage(response.modOnlyMessage)
-                                             .withAllowStartStopRecording(response.allowStartStopRecording).withBreakout(response.isBreakout)
+                                             .withAllowStartStopRecording(response.allowStartStopRecording)
+											 .withwebcamsOnlyForModerator(response.webcamsOnlyForModerator)
```
Comment (capilkey) [inline bigbluebutton-client/src/org/bigbluebutton/main/model/users/JoinService.as]: Should be withWebcams

File: bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/Constants.java [hunk 3]
```diff
@@ -31,6 +31,7 @@
   public static final String RECORDING                       = "recording";
   public static final String AUTO_START_RECORDING            = "auto_start_recording";
   public static final String ALLOW_START_STOP_RECORDING      = "allow_start_stop_recording";
+  public static final String WEBCAMS_ONLY_FOR_MODERTORS      = "webcams_only_for_moderators";
```
Comment (capilkey) [inline bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/Constants.java]: All the previous instances of the name have been OnlyForModerator, but this one is Moderators (with an 's')

File: bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/MessageToJson.java [hunk 4]
```diff
@@ -35,6 +35,7 @@ public static String createMeetingMessageToJson(CreateMeetingMessage msg) {
 		payload.put(Constants.DURATION, msg.duration);
 		payload.put(Constants.AUTO_START_RECORDING, msg.autoStartRecording);
 		payload.put(Constants.ALLOW_START_STOP_RECORDING, msg.allowStartStopRecording);
+		payload.put(Constants.WEBCAMS_ONLY_FOR_MODERTORS, msg.webcamsOnlyForModerator);
```
Comment (capilkey) [inline bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/MessageToJson.java]: Extra 's' at the end here too.

File: bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/converters/messages/CreateMeetingMessage.java [hunk 5]
```diff
@@ -1,39 +1,40 @@
 package org.bigbluebutton.api.messaging.converters.messages;
```
Comment (capilkey) [inline bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/converters/messages/CreateMeetingMessage.java]: Shouldn't need to change the indentation for the whole file to make the three additions you need.

File: bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/Constants.java (hunk 3 above)
Comment (capilkey) [inline bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/Constants.java]: All of the other instances leave off the "s" at the end. Should probably remove to match.

File: bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/MessageToJson.java (hunk 4 above)
Comment (capilkey) [inline bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/MessageToJson.java]: All of the other instances leave off the "s" at the end. Should probably remove to match.

File: bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/converters/messages/CreateMeetingMessage.java (hunk 5 above)
Comment (GhaziTriki) [inline bigbluebutton-web/src/java/org/bigbluebutton/api/messaging/converters/messages/CreateMeetingMessage.java]: The indentation should be back to the prior version in the last commit.

**Pull request merged**